### DISCUSSION TITLE
Question about a specific MySQL query via Budibase

### DISCUSSION THREAD

**Discussion by anuragbhatia** [Budibase/budibase#10186] [Help]
Hello, 

I have some data (putting a simpler example to explain issue I am facing)

|id|City|ix_id|
|---|----| -------|
|1  |SFO|100|
|2  |BOM|101|
|3  |SFO|101|

Now I need to design an interface which can compare these based on ix_id. Thus input should take ix_id's to compare and give non-common output. So output here can be:

```
Unique city for ix_id 100
 NONE


Unique city for ix_id 101
BOM 
```

I am able to achieve this using something like ```select ix_id,City from table WHERE ixi_id=100 AND ix_id NOT IN (select ix_id =101)``` and vice versa in MySQL directly. 

And what makes this complex is that I want to do this comparison for n number of items i.e say 10-20 or so. Would greatly appreciate if someone can throw light on what I should be using here with budibase. 

Thanks.

**Comment by ConorWebb96** [comment]
Hey @anuragbhatia,

A little confused by this question.

Are you trying to pull all unique records from your mysql table? If so it would be something like the below:

`
SELECT ix_id
FROM uniqueId
GROUP BY ix_id
HAVING COUNT(*) = 1;
`

If you are trying to only pull out the first unique Item of each ix_id, I would recommend using DISTINCT. Example below.

`
SELECT DISTINCT ix_id, name FROM uniqueId;
`

Lastly I would like to mention there where so many issues with the sql query you provided. It had typos, additional letters in variables incorrect clauses etc. Fixed version is below. I don't see any reason why excluding records in this way would cause issues!

`
SELECT ix_id FROM table WHERE ix_id=100 AND ix_id NOT IN (SELECT ix_id FROM table WHERE ix_id=101);
`

Hopefully this helps!
